v_orm

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:**
 v 0.2.2 711e309
**OS:** 
ubuntu 20.04

**What did you do?**

Test ORM generation
---
```
void main__App_lv1_gen(main__App* app) {
	sdb__Agent_user lv1 = (sdb__Agent_user){
		.id = 0,
		.name = _SLIT("一级代理"),
		.nice_name = _SLIT("张三"),
		.contacts = (string){.str=(byteptr)"", .is_lit=1},
		.addr = (string){.str=(byteptr)"", .is_lit=1},
		.total_num = 0,
		.mobile = _SLIT("18610001000"),
		.is_scale_alloc = true,
		.alloc_value = 20,
		.login_password = _SLIT("1234"),
		.in_come_total = 0,
		.balance = 0,
		.back_total = 0,
		.received_total = 0,
		.deduction = 0,
		.deduction_type = 0,
		.deduction_start_at = (string){.str=(byteptr)"", .is_lit=1},
		.deduction_end_at = (string){.str=(byteptr)"", .is_lit=1},
		.status = 0,
		.is_disable = 0,
		.open_id = (string){.str=(byteptr)"", .is_lit=1},
		.head_img_url = (string){.str=(byteptr)"", .is_lit=1},
		.agentlevel_id = (sdb__Agentlevel){.id = 0,.name = _SLIT("一级代理"),.level = ((int)(main__Proxy_level_lv_1)),.power_val = 255,},
		.cash_out_info_id = {0},
		.agent_priv_id = 0,
	};
	sqlite__DB_exec(app->db, _SLIT("delete from Agent_user where moblie='18610001000'"));
	
	// sql insert
	sqlite__DB _t1089 = app->db;
	sqlite3_stmt* _t1090 = sqlite__DB_init_stmt(_t1089, _SLIT("INSERT INTO `Agent_user` (`name`, `nice_name`, `contacts`, `addr`, `total_num`, `mobile`, `is_scale_alloc`, `alloc_value`, `login_password`, `in_come_total`, `balance`, `back_total`, `received_total`, `deduction_type`, `deduction_start_at`, `deduction_end_at`, `status`, `is_disable`, `open_id`, `head_img_url`, `agentlevel_id`, `cash_out_info_id`, `agent_priv_id`) values (?1, ?2, ?3, ?4, ?5, ?6, ?7, ?8, ?9, ?10, ?11, ?12, ?13, ?14, ?15, ?16, ?17, ?18, ?19, ?20, ?21, ?22, ?23)"));
	sqlite3_bind_text(_t1090, 1, lv1.name.str, lv1.name.len, 0);
	sqlite3_bind_text(_t1090, 2, lv1.nice_name.str, lv1.nice_name.len, 0);
	sqlite3_bind_text(_t1090, 3, lv1.contacts.str, lv1.contacts.len, 0);
	sqlite3_bind_text(_t1090, 4, lv1.addr.str, lv1.addr.len, 0);
	sqlite3_bind_int(_t1090, 5 , lv1.total_num); // stmt
	sqlite3_bind_text(_t1090, 6, lv1.mobile.str, lv1.mobile.len, 0);
	sqlite3_bind_int(_t1090, 7 , lv1.is_scale_alloc); // stmt
	sqlite3_bind_int(_t1090, 8 , lv1.alloc_value); // stmt
	sqlite3_bind_text(_t1090, 9, lv1.login_password.str, lv1.login_password.len, 0);
	sqlite3_bind_int(_t1090, 10 , lv1.in_come_total); // stmt
	sqlite3_bind_int(_t1090, 11 , lv1.balance); // stmt
	sqlite3_bind_int(_t1090, 12 , lv1.back_total); // stmt
	sqlite3_bind_int(_t1090, 13 , lv1.received_total); // stmt
	sqlite3_bind_int(_t1090, 14 , lv1.deduction_type); // stmt
	sqlite3_bind_text(_t1090, 15, lv1.deduction_start_at.str, lv1.deduction_start_at.len, 0);
	sqlite3_bind_text(_t1090, 16, lv1.deduction_end_at.str, lv1.deduction_end_at.len, 0);
	sqlite3_bind_int(_t1090, 17 , lv1.status); // stmt
	sqlite3_bind_int(_t1090, 18 , lv1.is_disable); // stmt
	sqlite3_bind_text(_t1090, 19, lv1.open_id.str, lv1.open_id.len, 0);
	sqlite3_bind_text(_t1090, 20, lv1.head_img_url.str, lv1.head_img_url.len, 0);
	
	// sql insert
	sqlite__DB _t1091 = app->db;
	sqlite3_stmt* _t1092 = sqlite__DB_init_stmt(_t1091, _SLIT("INSERT INTO `Agentlevel` (`name`, `level`, `power_val`) values (?1, ?2, ?3)"));
	sqlite3_bind_text(_t1092, 1, lv1.agentlevel_id.name.str, lv1.agentlevel_id.name.len, 0);
	sqlite3_bind_int(_t1092, 2 , lv1.agentlevel_id.level); // stmt
	sqlite3_bind_int(_t1092, 3 , lv1.agentlevel_id.power_val); // stmt
	
		int _t1093 = sqlite3_step(_t1092);
		if( (_t1093 != SQLITE_OK) && (_t1093 != SQLITE_DONE)){ puts(sqlite3_errmsg(_t1091.conn)); }
		sqlite3_finalize(_t1092);
	Array_sqlite__Row rows = sqlite__DB_exec(_t1089, _SLIT("SELECT last_insert_rowid()")).arg0;
	int _t1094 = string_int((*(string*)array_get((*(sqlite__Row*)array_get(rows, 0)).vals, 0)));
	sqlite3_bind_int(_t1090, 21 , _t1094); // id
	
	// sql insert
	sqlite__DB _t1095 = app->db;
	sqlite3_stmt* _t1096 = sqlite__DB_init_stmt(_t1095, _SLIT("INSERT INTO `Cash_out_info` (`we_chat`, `ali_pay`, `bank_card`, `bank_card_name`, `cash_out_total`, `cash_out_nums`, `we_chat_nice_name`, `we_chat_icon`, `cash_out_ty`) values (?1, ?2, ?3, ?4, ?5, ?6, ?7, ?8, ?9)"));
	sqlite3_bind_text(_t1096, 1, lv1.cash_out_info_id.we_chat.str, lv1.cash_out_info_id.we_chat.len, 0);
	sqlite3_bind_text(_t1096, 2, lv1.cash_out_info_id.ali_pay.str, lv1.cash_out_info_id.ali_pay.len, 0);
	sqlite3_bind_text(_t1096, 3, lv1.cash_out_info_id.bank_card.str, lv1.cash_out_info_id.bank_card.len, 0);
	sqlite3_bind_text(_t1096, 4, lv1.cash_out_info_id.bank_card_name.str, lv1.cash_out_info_id.bank_card_name.len, 0);
	sqlite3_bind_int(_t1096, 5 , lv1.cash_out_info_id.cash_out_total); // stmt
	sqlite3_bind_int(_t1096, 6 , lv1.cash_out_info_id.cash_out_nums); // stmt
	sqlite3_bind_text(_t1096, 7, lv1.cash_out_info_id.we_chat_nice_name.str, lv1.cash_out_info_id.we_chat_nice_name.len, 0);
	sqlite3_bind_text(_t1096, 8, lv1.cash_out_info_id.we_chat_icon.str, lv1.cash_out_info_id.we_chat_icon.len, 0);
	sqlite3_bind_int(_t1096, 9 , lv1.cash_out_info_id.cash_out_ty); // stmt
	
		int _t1097 = sqlite3_step(_t1096);
		if( (_t1097 != SQLITE_OK) && (_t1097 != SQLITE_DONE)){ puts(sqlite3_errmsg(_t1095.conn)); }
		sqlite3_finalize(_t1096);
	Array_sqlite__Row rows = sqlite__DB_exec(_t1089, _SLIT("SELECT last_insert_rowid()")).arg0;
	int _t1098 = string_int((*(string*)array_get((*(sqlite__Row*)array_get(rows, 0)).vals, 0)));
	sqlite3_bind_int(_t1090, 22 , _t1098); // id
	sqlite3_bind_int(_t1090, 23 , lv1.agent_priv_id); // stmt
	
		int _t1099 = sqlite3_step(_t1090);
		if( (_t1099 != SQLITE_OK) && (_t1099 != SQLITE_DONE)){ puts(sqlite3_errmsg(_t1089.conn)); }
		sqlite3_finalize(_t1090);
}
```

**What did you expect to see?**

normal
 
**What did you see instead?**
```
/tmp/v/v.692643950402262625.tmp.c:23807: warning: assignment discards qualifiers from pointer target type
/tmp/v/v.692643950402262625.tmp.c:23813: warning: assignment discards qualifiers from pointer target type
/tmp/v/v.692643950402262625.tmp.c:23817: warning: assignment discards qualifiers from pointer target type
/tmp/v/v.692643950402262625.tmp.c:23987: error: redeclaration of 'rows'
builder error: 
==================
C error. This should never happen.

If you were not working with C interop, this is a compiler bug, please raise an issue on GitHub:

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

--- a/vlib/v/gen/c/sql.v
+++ b/vlib/v/gen/c/sql.v
@@ -150,9 +150,10 @@ fn (mut g Gen) sqlite3_stmt(node ast.SqlStmt, typ SqlType) {
 				g.sql_stmt_name = tmp_sql_stmt_name
 				g.sql_table_name = tmp_sql_table_name
 				// get last inserted id
-				g.writeln('Array_sqlite__Row rows = sqlite__DB_exec($db_name, _SLIT("SELECT last_insert_rowid()")).arg0;')
+				res := g.new_tmp_var()
+				g.writeln('Array_sqlite__Row $res = sqlite__DB_exec($db_name, _SLIT("SELECT last_insert_rowid()")).arg0;')
 				id_name := g.new_tmp_var()
-				g.writeln('int $id_name = string_int((*(string*)array_get((*(sqlite__Row*)array_get(rows, 0)).vals, 0)));')
+				g.writeln('int $id_name = string_int((*(string*)array_get((*(sqlite__Row*)array_get($res, 0)).vals, 0)));')
 				g.writeln('sqlite3_bind_int($g.sql_stmt_name, ${i + 0} , $id_name); // id')
 			} else {
 				g.writeln('sqlite3_bind_int($g.sql_stmt_name, ${i + 0} , $x); // stmt')

